fe/fe-core/src/main/java/org/apache/doris/load/loadv2/MysqlLoadManager.java

```diff
@@ -43,55 +59,297 @@
+                            LOG.warn("Execute stream load for mysql data load failed with message: " + request);
+                            throw new LoadException(result.get("Message").getAsString());
+                        }
+                        loadResult.incRecords(result.get("NumberLoadedRows").getAsLong());
+                        loadResult.incSkipped(result.get("NumberFilteredRows").getAsInt());
                     }
-                    loadResult.incRecords(result.get("NumberLoadedRows").getAsLong());
-                    loadResult.incSkipped(result.get("NumberFilteredRows").getAsInt());
                 }
             }
+            releaseToken(context, token);
+        } else {
+            StreamLoadTxnExecutor executor = null;
+            try {
+                String database = dataDesc.getFullDatabaseName();
+                String table = dataDesc.getTableName();
+                TransactionEntry entry = prepareTransactionEntry(database, table);
+                openTxn(context, entry);
+                executor = beginTxn(context, entry, dataDesc);
+                // sendData
+                for (String file : filePaths) {
+                    sendData(context, executor, file);
+                }
+                executor.commitTransaction();
+            } catch (Exception e) {
+                LOG.error("Failed to load mysql data into doris", e);
+                if (executor != null) {
+                    try {
+                        executor.abortTransaction();
+                    } catch (Exception ex) {
+                        throw new LoadException("Failed when abort the transaction", ex);
+                    }
+                }
+                throw new LoadException("Load failed when execute the mysql data load", e);
+            }
         }
         return loadResult;
     }
 
+    private TransactionEntry prepareTransactionEntry(String database, String table)
+            throws TException {
+        TTxnParams txnConf = new TTxnParams();
+        txnConf.setNeedTxn(true).setEnablePipelineTxnLoad(Config.enable_pipeline_load)
+                .setThriftRpcTimeoutMs(5000).setTxnId(-1).setDb("").setTbl("");
+        Database dbObj = Env.getCurrentInternalCatalog()
+                .getDbOrException(database, s -> new TException("database is invalid for dbName: " + s));
+        Table tblObj = dbObj.getTableOrException(table, s -> new TException("table is invalid: " + s));
+        txnConf.setDbId(dbObj.getId()).setTbl(table).setDb(database);
+
+        TransactionEntry txnEntry = new TransactionEntry();
+        txnEntry.setTxnConf(txnConf);
+        txnEntry.setTable(tblObj);
+        txnEntry.setDb(dbObj);
+        String label = UUID.randomUUID().toString();
+        txnEntry.setLabel(label);
+        return txnEntry;
+    }
+
+    private void openTxn(ConnectContext context, TransactionEntry txnEntry) throws Exception {
```

Comment:
The method name is confusing.
in `openTxn()`, it actually "beginTransaction()".
and in `beginTxn()`, it actually do the plan job.